tests/js/spec/views/dashboardsV2/widgetBuilder/widgetBuilder.spec.tsx

```diff
@@ -818,6 +1071,31 @@ describe('WidgetBuilder', function () {
         );
       });
     });
+
+    it('disables dashboards with max widgets', async function () {
+      MockApiClient.addMockResponse({
+        url: '/organizations/org-slug/dashboards/',
+        body: [
+          {...untitledDashboard, widgetDisplay: []},
+          {...testDashboard, widgetDisplay: [DisplayType.TABLE]},
+        ],
+      });
+
+      Object.defineProperty(dashboardsTypes, 'MAX_WIDGETS', {value: 1});
+
+      renderTestComponent({
+        query: {
+          source: DashboardWidgetSource.DISCOVERV2,
+        },
+      });
+
+      userEvent.click(await screen.findByText('Select a dashboard'));
+      userEvent.type(screen.getByText('Select a dashboard'), 'Test Dashboard{enter}');
+
+      // Dashboard wasn't selected because it has the max number of widgets
```

Comment:
I had to swap out the tooltip check for "Max widgets (1) per dashboard reached." because it was beginning to flaky both locally and on CI, instead I opted for checking that the behaviour of trying to select it by filtering the list and hitting enter doesn't set the field